### PR TITLE
Option to show messages in tests.

### PR DESCRIPTION
**Pull request by chriseth**
If you use `soltest -- --show-messages` it will print the call and return data of each transaction during testing.

### CODE REVIEW

File: test/ExecutionFramework.cpp [hunk 1]
```diff
@@ -46,13 +46,21 @@ string getIPCSocketPath()
...
 }
 
 void ExecutionFramework::sendMessage(bytes const& _data, bool _isCreation, u256 const& _value)
 {
+	if (m_showMessages)
+	{
+		if (_isCreation)
+			cout << "CREATE " << toHex(m_sender) << ": " << toHex(_data) << endl;
+		else
+			cout << "CALL   " << toHex(m_sender) << " -> " << toHex(m_contractAddress) << ": " << toHex(_data) << endl;
```
Comment (axic) [inline test/ExecutionFramework.cpp]: Can we display the tx value (if non-zero) for both create/call?

File: test/ExecutionFramework.cpp [hunk 2]
```diff
@@ -79,6 +87,9 @@ void ExecutionFramework::sendMessage(bytes const& _data, bool _isCreation, u256
 		m_output = fromHex(code, WhenError::Throw);
 	}
 
+	if (m_showMessages)
+		cout << " -> " << toHex(m_output) << endl;
```
Comment (axic) [inline test/ExecutionFramework.cpp]: As a coding style, shouldn't we use `std::` with these? (I know in this file the std namespace is imported.)

Comment (chriseth) [reply test/ExecutionFramework.cpp]: There should be no `std::` in cpp files.

**Review by axic — APPROVED** [review]

**Comment by axic** [comment]
Restarted the tests, but they've failed in different locations in each of the three runs under clang (under gcc it worked):
```
/home/travis/build/ethereum/solidity/test/libsolidity/SolidityEndToEndTest.cpp(5180): error in "return_multiple_strings_of_various_sizes": check callContractFunction("s1()") == encodeArgs(0x20) + dyn1 failed

/home/travis/build/ethereum/solidity/test/libsolidity/SolidityEndToEndTest.cpp(1131): error in "structs": check callContractFunction("check()") == encodeArgs(true) failed

/home/travis/build/ethereum/solidity/test/contracts/Wallet.cpp(598): fatal error in "revoke_addOwner": critical check callContractFunction("isOwner(address)", h256(0x33)) == encodeArgs(false) failed
```

Do we potentially have some bug with clang build?

**Comment by chriseth** [comment]
@axic are you sure this is not a problem in the backend? Do you get "mining" errors?

**Comment by axic** [comment]
There were no mining errors this time. `eth` has its own problems as frequently there are mining errors or "Account not found".

**Comment by chriseth** [comment]
Just tried clang-3.5, but I did not get any failures.

**Comment by chriseth** [comment]
Merging since I cannot reproduce the errors you mentioned. Also note that travis has a clang run which is also passing.

**Comment by axic** [comment]
It was failing on travis as you see the logs above, I've restarted it after it failed.

**Pull request merged**